normandy/selfrepair/static/js/self_repair_driver.js

```diff
@@ -0,0 +1,132 @@
+/**
+ * Convenience wrapper around XMLHttpRequest.
+ *
+ * @param {String} method - HTTP method to use.
+ * @param {String} url - URL to send the request to.
+ * @param {Object} options - Optional arguments.
+ * @param {Object} options.data - Request data as an Object; keys are
+ *     used as parameter names, values as parameter values.
+ * @param {Object} options.headers - Headers and values to set on the
+ *     request.
+ * @promise {XMLHttpRequest} The request after it has succeeded.
+ * @rejects {XMLHttpRequest} The request after it has failed.
+ */
+function xhr(method, url, options={}) {
+    return new Promise((resolve, reject) => {
+        let req = new XMLHttpRequest();
+        req.addEventListener('load', e => {
+            if (req.status !== 200) {
```

Comment:
This should be something like `if (req.status >= 400) {`. [201][] and [204][] are totally useful status codes.

[201]: http://http.cat/201
[204]: http://http.cat/204